core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -492,20 +498,25 @@ def _convert_proba_to_unified_form(self, y_pred_proba):
     def score(self, X, y, metric=None, sample_weight=None, **kwargs):
         if metric is None:
             metric = self.eval_metric
+
         if metric.needs_pred:
             y_pred = self.predict(X=X, **kwargs)
+        elif metric.needs_quantile:
+            y_pred = self.predict(X=X, **kwargs)
         else:
             y_pred = self.predict_proba(X=X, **kwargs)
-        return compute_weighted_metric(y, y_pred, metric, sample_weight)
+        return compute_weighted_metric(y, y_pred, metric, sample_weight, quantile_levels=self.quantile_levels)
 
     def score_with_y_pred_proba(self, y, y_pred_proba, metric=None, sample_weight=None):
         if metric is None:
             metric = self.eval_metric
         if metric.needs_pred:
             y_pred = get_pred_from_proba(y_pred_proba=y_pred_proba, problem_type=self.problem_type)
+        elif metric.needs_quantile:
```

Comment:
`or` is cleaner.

Side note: Isn't this the same as `y_pred = y_pred_proba`? Why not just include it in the `else` logic? and avoid changing the code at all?

Reply:
This will be fixed by just removing it.